src/features/locators/components/LocatorEditDialog.tsx

```diff
@@ -201,6 +200,44 @@ export const LocatorEditDialog: React.FC<Props> = ({
       </div>
     ) : null;
 
+  const onLocatorTypeChange = (value: LocatorType) => {
+    setLocatorType(value);
+    const isPrevLocatorValid = isValidLocator(message);
+    const isNewLocatorValid = isValidLocator(validationMessage);
+    const isCSSLocator = value === LocatorType.cssSelector;
+
+    // check and rework this condition after css locator enabling
+    if (form.isFieldTouched("locator")) {
+      if (isPrevLocatorValid) {
+        if (isNewLocatorValid) {
+          const customXpath = isCSSLocator ? form.getFieldValue("locator") : locatorField;
+          form.setFieldValue("locator", getLocator({ ...locator, customXpath }, value));
+        } else {
+          form.setFieldValue(
+            "locator",
+            isCSSLocator ? getLocator(locator, value) : getLocator({ ...locator, customXpath: locatorField }, value)
+          );
+        }
+      } else {
+        form.setFieldValue(
+          "locator",
+          isNewLocatorValid ? getLocator({ ...locator, customXpath: form.getFieldValue("locator") }, value) : ""
+        );
+      }
+    } else if (!isPrevLocatorValid) {
+      value === locatorType ? form.setFieldValue("locator", locatorField) : form.setFieldValue("locator", "");
+    } else {
+      form.setFieldValue("locator", getLocator(locator, value));
+    }
+  };
+
+  const onFieldsChange = async (changedValues: any) => {
+    const [changedValue] = changedValues;
+    const isLocatorTypeChanged = changedValue?.name[0] === "locatorType";
```

Comment:
what is the type of `changedValues`? Is there always will be inly one value? Using index can be unstable for cases when there are more that one item in array

Reply:
fixed